packages/create-cloudflare/src/helpers/command.ts

```diff
@@ -289,3 +314,28 @@ export const listAccounts = async () => {
 
 	return accounts;
 };
+
+/**
+ * Look up the latest release of workerd and use its date as the compatibility_date
+ * configuration value for wrangler.toml.
+ *
+ * If the look up fails then we fall back to a well known date.
+ *
+ * The date is extracted from the version number of the workerd package tagged as `latest`.
+ * The format of the version is `major.yyyymmdd.patch`.
+ *
+ * @returns The latest compatibility date for workerd in the form "YYYY-MM-DD"
+ */
+export async function getWorkerdCompatibilityDate() {
+	return runCommand("npm info workerd dist-tags.latest", {
```

Comment:
Out of interest, why run a command here instead of just hitting the npm API directly? That seems safer and more reliable to me? 

`https://registry.npmjs.org/workerd`

Reply:
Thanks for the suggestion @Cherry - @mrbbot also suggested this during an in-person review.
 
My take was:
- all these templates will require npm commands to run, so if npm is broken then you have bigger problems, so reliability seems irrelevant
- I am not sure what is meant by safer here?
- It is possible that hitting npm REST directly might be faster, but I doubt that it would be notable in the bigger picture of generating a project

I am also not clear how public and reliable the npm REST API is. I can't find any official (non-archived) docs for the API. Finally, using the CLI directly could benefit from accessing the cached packages (totally guessing here!) which actually might be faster than hitting the API??

So, this is working right now. I am happy to change it if there is enough evidence or concern about this approach.

Reply:
Thanks for the detailed thoughts @petebacondarwin!

My comment about being safer was simply due to avoiding needing to spin up a new process, etc. and the overhead that comes with that. I know it's technically possible to run `node` without `npm` even being installed, but as you noted, almost all of these frameworks and templates recommended/require `npm` in some capacity anyway so this probably isn't a huge concern.

For the npm API, I don't think they have any official public docs outside of the packages they publish to interact with it, such as https://github.com/npm/pacote and https://github.com/npm/npm-registry-fetch, etc. `npm info` also seems to be an alias for `npm view`, and it doesn't seem to use `cache` at all from what I can see at https://github.com/npm/cli/blob/latest/lib/commands/view.js, and just hits the npm API directly using `pacote`.

As you mentioned though, this is probably _fine_ and not a huge concern. 👍